src/main/java/org/opentripplanner/middleware/auth/Auth0Users.java

```diff
@@ -242,33 +248,39 @@ private static String getAuth0Url() {
     }
 
     /**
-     * Get an Auth0 oauth token for use in mocking user requests by using the Auth0 'Call Your API Using Resource Owner
-     * Password Flow' approach. Auth0 setup can be reviewed here: https://auth0.com/docs/flows/call-your-api-using-resource-owner-password-flow.
-     * If the user is successfully validated by Auth0 a bearer access token is returned, which is extracted and returned
-     * to the caller. In all other cases, null is returned.
+     * Get an Auth0 oauth token response for use in mocking user requests by using the Auth0 'Call Your API Using Resource
+     * Owner Password Flow' approach. Auth0 setup can be reviewed here: https://auth0.com/docs/flows/call-your-api-using-resource-owner-password-flow.
+     * If token response is returned to calling methods for evaluation.
      */
-    public static String getAuth0Token(String username, String password) throws JsonProcessingException {
+    public static HttpResponse<String> getCompleteAuth0TokenResponse(String username, String password) {
         if (Auth0Connection.isAuthDisabled()) return null;
         String body = String.format(
             "grant_type=password&username=%s&password=%s&audience=%s&scope=&client_id=%s&client_secret=%s",
             username,
             password,
-            "https://otp-middleware", // must match an API identifier
-            AUTH0_CLIENT_ID, // Auth0 application client ID
-            AUTH0_CLIENT_SECRET // Auth0 application client secret
+            DEFAULT_AUDIENCE, // must match an API identifier
+            (isEndToEnd) ? AUTH0_CLIENT_ID : AUTH0_API_CLIENT, // Auth0 application client ID
+            (isEndToEnd) ? AUTH0_CLIENT_SECRET : AUTH0_API_SECRET // Auth0 application client secret
```

Comment:
What is this check for `isEndToEnd`? I can't fully comment on this because I don't know what it's doing, but this doesn't seem right. From what I remember, the `AUTH0_API_SECRET` is used for getting an API token for performing management actions (not for an end user/standard Auth0 user).

Reply:
@landonreed before I make any updates I would like to define the correct approach. The reason for this is because CI (as I understand it) does not have AUTH_API_CLIENT nor AUTH_API_SECRET, hence the check. But in my conf AUTH0_CLIENT_ID = AUTH0_API_CLIENT and AUTH0_CLIENT_SECRET = AUTH0_API_SECRET. It might just be a case of removing this check and adding AUTH_API_CLIENT and AUTH_API_SECRET to the CI conf (even if the values match AUTH0_CLIENT_ID and AUTH0_CLIENT_SECRET as they do for me locally).